src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/BaseHeartbeatProperties.cs

```diff
@@ -0,0 +1,181 @@
+            return this.DefaultFields.Contains(keyword, StringComparer.OrdinalIgnoreCase);
+        }
+
+        public Task<bool> SetDefaultPayload(IEnumerable<string> disabledFields, IHeartbeatProvider provider)
+        {
+            bool hasSetValues = false;
+            var enabledProperties = this.DefaultFields.Except(disabledFields);
+
+            foreach (string fieldName in enabledProperties)
+            {
+                // we don't need to report out any failure here, so keep this look within the Sdk Internal Operations as well
+                try
+                {
+                    switch (fieldName)
+                    {
+                        // case "runtimeFramework":
+                        //    provider.AddHeartbeatProperty(fieldName, true, this.GetRuntimeFrameworkVer(), true);
+                        //    hasSetValues = true;
+                        //    break;
+                        case "baseSdkTargetFramework":
+                            provider.AddHeartbeatProperty(fieldName, true, this.GetBaseSdkTargetFramework(), true);
+                            hasSetValues = true;
+                            break;
+                        case "osType":
+                            provider.AddHeartbeatProperty(fieldName, true, this.GetRuntimeOsType(), true);
+                            hasSetValues = true;
+                            break;
+                        case "processSessionId":
+                            provider.AddHeartbeatProperty(fieldName, true, this.GetProcessSessionId(), true);
+                            hasSetValues = true;
+                            break;
+                        default:
+                            provider.AddHeartbeatProperty(fieldName, true, "UNDEFINED", true);
+                            break;
+                    }
+                }
+                catch (Exception ex)
+                {
+                    CoreEventSource.Log.FailedToObtainDefaultHeartbeatProperty(fieldName, ex.ToString());
+                }
+            }
+
+            return Task.FromResult(hasSetValues);
+        }
+
+        // NOTE: We would like to include the NETSTANDARD/NETCORE/NETFRAMEWORK version in the core heartbeat 
+        // but until we understand how to make the mapping from the runtime version to the standard versions
+        // this information is not entirely useful. Leaving it commented out here for now until such a mapping
+        // can be made in an appropriate manner, or until we can provide such a mapping.
+        //
+        // <summary>
+        // This will return the current running .NET framework version, based on the version of the assembly that owns
+        // the 'Object' type. The version number returned can be used to infer other things such as .NET Core / Standard.
+        // </summary>
+        // <returns>a string representing the version of the current .NET framework</returns>
+        // private string GetRuntimeFrameworkVer()
```

Comment:
I just realized that this code is already shipping in 2.5.0 stable. We should try to find this mapping ourselves, and post it in Github or some place for quick reference. @Dmitry-Matveev also to comment on best way to support already shipped (2.5) sdk.

Reply:
Having a reference list of the known combinations of base .NET version and runtime version (in the way we retrieve it in 2.5) should make this feature more useful for our customers as well as for internal investigations. We may want to create such a list for 2.5.

Reply:
I feel that this discussion needs to be sorted out separately from this PR (it's related, but tangential to the scope of this discussion somewhat). I've raised a new issue (#699) where we can follow up and deal with this in a specific place. 

I've re-enabled the code and left it as it was in 2.5.0-stable for now.